### ISSUE TITLE
[0.8.1.4 beta] Server exception on harvesting plants under fish law

### ISSUE THREAD

**Issue by lashtear** [StrangeLoopGames/EcoIssues#12413]
Recently enabled a new law:

When attempting to Craft where (item is Fish Trap)
or Craft where (item is Fishing Pole)
or Harvest where (item is Trout)
or Harvest where (item is Salmon)
or Harvest where (item is Tuna)
prevent

But now harvesting e.g. plants fails with a server-exception and boots the user:

Server encountered an exception:

```
Exception: InvalidCastException
Message:Unable to cast object of type 'Eco.Gameplay.Stats.ConcretePlayerActions.HarvestAction' to type 'Eco.Stats.IHasItemContext'.
Source:Eco.Gameplay

System.InvalidCastException: Unable to cast object of type 'Eco.Gameplay.Stats.ConcretePlayerActions.HarvestAction' to type 'Eco.Stats.IHasItemContext'.
   at Eco.Gameplay.LegislationSystem.LawLogic.ActionQueries.ItemActionQuery.IsSelected(IPlayerAction action)
   at System.Linq.Enumerable.Any[TSource](IEnumerable`1 source, Func`2 predicate)
   at Eco.Gameplay.LegislationSystem.LawLogic.Clauses.ActionList.IsAllowed(IPlayerAction action)
   at System.Linq.Enumerable.WhereSelectEnumerableIterator`2.MoveNext()
   at System.Linq.Enumerable.WhereEnumerableIterator`1.MoveNext()
   at System.Linq.Enumerable.Any[TSource](IEnumerable`1 source)
   at Eco.Gameplay.LegislationSystem.LawLogic.LawLogicRoot.IsAllowed(IPlayerAction action)
   at Eco.Gameplay.Legislation.CreateAtomicAction(IPlayerAction action)
   at System.Linq.Enumerable.WhereSelectArrayIterator`2.MoveNext()
   at System.Collections.Generic.List`1.InsertRange(Int32 index, IEnumerable`1 collection)
   at Eco.Gameplay.Stats.PlayerActionManager`1.CreateAtomicAction(T action)
   at Eco.Gameplay.Stats.ConcretePlayerActions.HarvestPlayerActionManager.CreateAtomicAction(User actor, Organism target)
   at Eco.Gameplay.Plants.PlantEntity.CanHarvest(Player player, Boolean picked)
   at Eco.Gameplay.Plants.PlantEntity.TryHarvest(Player player, Boolean picked, Inventory targetInv)
   at Eco.Mods.TechTree.ScytheItem.OnActLeft(InteractionContext context)
   at Eco.Gameplay.Interactions.InteractionExtensions.ExecuteUntilSuccess(IEnumerable`1 results)
   at Eco.Gameplay.Players.Player.PlayerInteract(InteractionInfo info)</size>
```

**Comment by SlayksWood** [comment]
8.2.0 staging-14-n
First type law:
![image](https://user-images.githubusercontent.com/45708377/58411691-84df3a00-807d-11e9-97cf-168cfe195716.png)
Second type law:
![image](https://user-images.githubusercontent.com/45708377/58412941-3bdcb500-8080-11e9-9eca-c2bf92df31bc.png)
Both laws do not allow me to craft a fishing pole or trap. But in two laws, I can fish.
When collecting plants, digging, mining, etc., I did not find anything strange.
![image](https://user-images.githubusercontent.com/45708377/58411427-f1a60480-807c-11e9-94d6-235a0835d941.png)
![image](https://user-images.githubusercontent.com/45708377/58412989-5a42b080-8080-11e9-986c-933c94b0f504.png)

Third type law:
![image](https://user-images.githubusercontent.com/45708377/58413761-4009d200-8082-11e9-9a26-932114923f45.png)
![image](https://user-images.githubusercontent.com/45708377/58413542-b528d780-8081-11e9-9b82-6669a5fba8fb.png)
I can't propose law which prohibits fishing.

**Comment by SlayksWood** [comment]
@lashtear can your provide save files with law?
Instruction is here https://github.com/StrangeLoopGames/EcoIssues/wiki/Uploading-a-world-save

**Comment by SlayksWood** [comment]
If you shoot in Tuna it work. But you can catch fish with Fishing Pole and Fish Trap
![image](https://user-images.githubusercontent.com/45708377/58415001-c7a51000-8085-11e9-8426-b36850b51b60.png)

**Comment by lashtear** [comment]
The law text they wrote is as above.  I agree that they should not have been able to enter that-- I'm thinking a race condition in the web app.  Note that it says Harvest-where-ITEM-is ... In local testing I get at the top of the law editor page:

Error: Internal Error: Law node 'Eco.Gameplay.LegislationSystem.LawLogic.ActionQueries.ItemActionQuery' missing required environment variable 'Action.IHasItemContext'

(buglet: very hard to select that text)

I'm not owner or admin on the original server, so I can't produce save files of it, unfortunately.  Version supplied is local-client.  I don't know what web browser the law author was using.  I've asked him, though!

**Comment by mirasrael** [comment]
Fixed in 8.2.2
